### PR TITLE
bext: include integration tests snapshots into client Percy build

### PR DESCRIPTION
**Pull request by taras-yemets**
Combines browser extension and web snapshots into a single Percy build.

It was initially added in https://github.com/sourcegraph/sourcegraph/pull/31555/, but then reverted as one of the tests appeared to be flaky. It took some time to fix the browser extension integration tests setup (https://github.com/sourcegraph/sourcegraph/pull/33598), and new changes to how Percy build is finalized were made in https://github.com/sourcegraph/sourcegraph/pull/33498. Re-adding browser extension Percy snapshots to the final Percy build.

## Test plan
Check the pipelines on the client code PRs and ensure that Percy pipeline result includes web and browser extension snapshots.

### CODE REVIEW

File: enterprise/dev/ci/internal/ci/pipeline.go [hunk 1]
```diff
@@ -121,7 +121,8 @@ func GeneratePipeline(c Config) (*bk.Pipeline, error) {
 		// builds.
 		ops = operations.NewSet(
 			addClientLinters,
-			addBrowserExt,
+			browserUnitTests,
+			browserIntegrationTests(0),
```
[the remark this answers is not in the document]
Comment (taras-yemets) [reply enterprise/dev/ci/internal/ci/pipeline.go]: Thanks for your review, @jhchabran!
To date, I don't expect to increase `parallelTestCount` value for `BextReleaseBranch` and `BextNightly` pipelines in future. We don't run web integration tests for these two pipelines, so we don't expect any other steps to make Percy snapshots which we want to see in the resulting Percy build. 
Unlike the case with [client integration tests](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@taras-yemets/include-bext-snapshots-into-percy-build/-/blob/enterprise/dev/ci/internal/ci/operations.go?L255-294&utm_source=VSCode-2.2.0) on the PR. There we run web and browser extension integration tests and want to see all Percy snapshots in the resulting build. So we calculate the number of steps contributing to this build and pass it as the `PERCY_PARALLEL_TOTAL` environment variable value.
Added a comment about it ([commit](https://github.com/sourcegraph/sourcegraph/pull/33709/commits/ab5c2bb93b497b5e449bd9b137dfb41402ee4229)).
This doesn't seem to be an elegant solution for this case, but I couldn't come up with anything better myself 🤷🏻‍♂️

File: client/browser/src/integration/gitlab.test.ts [hunk 2]
```diff
@@ -41,37 +41,42 @@ describe('GitLab', () => {
...
-                    })
+        // mock file blob response in replay mode: https://github.com/sourcegraph/sourcegraph/pull/33598#discussion_r846137281
+        if (readEnvironmentString({ variable: 'POLLYJS_MODE', defaultValue: 'replay' }) === 'replay') {
+            testContext.server
+                .get(
+                    'https://gitlab.com/sourcegraph/jsonrpc2/-/blob/4fb7cd90793ee6ab445f466b900e6bffb9b63d78/call_opt.go?format=json&viewer=simple'
```
Comment (taras-yemets) [inline client/browser/src/integration/gitlab.test.ts:48]: Thank you for [your comment](https://github.com/sourcegraph/sourcegraph/pull/33598#discussion_r848972750), @tjkandala! `POLLYJS_MODE` value check before mocking the response should cover this case 🚀

File: enterprise/dev/ci/internal/ci/pipeline.go [hunk 3]
```diff
@@ -121,7 +121,8 @@ func GeneratePipeline(c Config) (*bk.Pipeline, error) {
 		// builds.
 		ops = operations.NewSet(
 			addClientLinters,
-			addBrowserExt,
+			browserUnitTests,
+			browserIntegrationTests(0), // we pass 0 here as we don't other pipeline steps to contribute to the resulting Percy build
```
Comment (erzhtor) [inline enterprise/dev/ci/internal/ci/pipeline.go:125]:
```suggestion
			browserIntegrationTests(0), // we pass 0 here as we don't have other pipeline steps to contribute to the resulting Percy build
```

File: enterprise/dev/ci/internal/ci/pipeline.go [hunk 4]
```diff
@@ -131,7 +132,8 @@ func GeneratePipeline(c Config) (*bk.Pipeline, error) {
 		// e2e tests.
 		ops = operations.NewSet(
 			addClientLinters,
-			addBrowserExt,
+			browserUnitTests,
+			browserIntegrationTests(0), // we pass 0 here as we don't other pipeline steps to contribute to the resulting Percy build
```
Comment (erzhtor) [inline enterprise/dev/ci/internal/ci/pipeline.go:136]:
```suggestion
			browserIntegrationTests(0), we pass 0 here as we don't have other pipeline steps to contribute to the resulting Percy build
```

File: enterprise/dev/ci/internal/ci/operations.go [hunk 5]
```diff
@@ -210,28 +210,38 @@ func addWebApp(pipeline *bk.Pipeline) {
...
+				bk.Cmd("yarn nyc report -r json"),
+				bk.Cmd("dev/ci/codecov.sh -c -F typescript -F integration"),
+				bk.ArtifactPaths("./puppeteer/*.png"),
+			)
+		}
 	}
+}
 
-	// Browser extension unit tests
+func browserUnitTests(pipeline *bk.Pipeline) {
```
Comment (erzhtor) [inline enterprise/dev/ci/internal/ci/operations.go:244]: WDYT to explicitly include browserExtension or browserExt in the function name?
Same for the `browserUnitTests`, I think.

Comment (erzhtor) [reply enterprise/dev/ci/internal/ci/operations.go:244]: Also would be great to keep consistent with the rest of the pipeline contribution steps (like `addBrowserExtensionReleaseSteps`): `addBrowserExtensionUnitTests, addBrowserExtensionIntegrationTests`.

**Review by erzhtor — APPROVED** [review]
This is awesome! Thank you for tackling and adding Browser Extension snapshot tests! 👍

**Review by vdavid — APPROVED** [review]
The code looks good to me.
I've also checked the Percy pipeline result on the [PR page](https://github.com/sourcegraph/sourcegraph/pull/33709/), and I saw that it included web and browser extension snapshots.

**Pull request merged**